core/templates/components/entity-creation-services/story-creation.service.ts

```diff
@@ -16,93 +16,72 @@
-          let postData = {
-            title: newlyCreatedStory.title,
-            description: newlyCreatedStory.description,
-            story_url_fragment: newlyCreatedStory.urlFragment,
-            thumbnailBgColor: bgColor,
-            filename: imagesData[0].filename
-          };
+  createNewCanonicalStory(): void {
+    if (this.storyCreationInProgress) {
+      return;
+    }
+    let modalRef = this.ngbModal.open(CreateNewStoryModalComponent, {
+      backdrop: 'static'
+    });
 
-          let body = new FormData();
-          body.append('payload', JSON.stringify(postData));
-          body.append('image', imagesData[0].imageBlob);
-
-          $http.post(createStoryUrl, (body), {
-            // The actual header to be added for form-data is
-            // 'multipart/form-data', But adding it manually won't work because
-            // we will miss the boundary parameter. When we keep 'Content-Type'
-            // as undefined the browser automatically fills the boundary
-            // parameter according to the form
-            // data. Refer https://stackoverflow.com/questions/37039852/. and
-            // https://stackoverflow.com/questions/34983071/.
-            // Note: This should be removed and a convetion similar to
-            // StoryCreationService should be followed once this service
-            // is migrated to Angular 8.
-            headers: {
-              'Content-Type': undefined
-            }
-          })
-            .then(function(response) {
-              $window.location = UrlInterpolationService.interpolateUrl(
-                STORY_EDITOR_URL_TEMPLATE, {
-                  story_id: response.data.storyId
-                }
-              );
-            }, function() {
-              LoaderService.hideLoadingScreen();
-              ImageLocalStorageService.flushStoredImagesData();
-            });
-        }, function() {
-          // Note to developers:
-          // This callback is triggered when the Cancel button is clicked.
-          // No further action is needed.
-        });
+    modalRef.result.then((newlyCreatedStory) => {
+      if (!newlyCreatedStory.isValid()) {
+        throw new Error('Story fields cannot be empty');
       }
-    };
+      this.storyCreationInProgress = true;
+      this.alertsService.clearWarnings();
+      this.loaderService.showLoadingScreen('Creating story');
+      let newTab = this.windowRef.nativeWindow;
+      let imagesData = this.imageLocalStorageService.getStoredImagesData();
+      let bgColor = this.imageLocalStorageService.getThumbnailBgColor();
+      this.storyCreationBackendApiService.createStoryAsync(
+        newlyCreatedStory, imagesData, bgColor).then((response) => {
+        this.storyCreationInProgress = false;
+        newTab.location.href = this.urlInterpolationService.interpolateUrl(
+          this.STORY_EDITOR_URL_TEMPLATE, {
+            story_id: response.storyId
```

Comment:
This used to be `response.data.storyId`. Can you make sure it should be changed to `response.storyId`?

Reply:
I have declared the interface in the story-creation-backend-api file named "StoryCreationResponse" from that we can directly access the storyId, so we don't need to put data in between.